pkg/gardenlet/controller/seed/seed/reconciler_reconcile.go

```diff
@@ -1041,19 +1042,28 @@ func (r *Reconciler) runReconcileSeedFlow(
 		return flow.Errors(err)
 	}
 
-	// Patching fluent-bit service with the network policy annotations
-	fluentBitService := &corev1.Service{}
-	if err := seedClient.Get(ctx, kubernetesutils.Key(v1beta1constants.GardenNamespace, v1beta1constants.DaemonsetNameFluentBit), fluentBitService); err != nil {
-		return err
-	}
-
-	fluentBitServicePatch := client.MergeFrom(fluentBitService.DeepCopy())
-	fluentBitService.ObjectMeta.Annotations[v1beta1constants.AnnotationNetworkPolicyFromPolicyPodLabelSelector] = v1beta1constants.NetworkPolicyAllScrapeTargets
-	fluentBitService.ObjectMeta.Annotations[v1beta1constants.AnnotationNetworkPolicyFromPolicyAllowedPorts] = "[{\"port\":\"2020\",\"protocol\":\"TCP\"},{\"port\":\"2021\",\"protocol\":\"TCP\"}]"
+	if loggingEnabled {
+		// Waiting for fluent-bit service to be created
+		fluentBitService := &corev1.Service{}
+		if err := retry.UntilTimeout(ctx, 5*time.Second, time.Minute, func(ctx context.Context) (done bool, err error) {
+			if err = seedClient.Get(ctx, kubernetesutils.Key(v1beta1constants.GardenNamespace, v1beta1constants.DaemonsetNameFluentBit), fluentBitService); err != nil {
+				return retry.MinorError(fmt.Errorf("%s service is not ready: %v", v1beta1constants.DaemonsetNameFluentBit, err))
```

Comment:
"does not exist" != "is not ready"

```suggestion
				return retry.MinorError(fmt.Errorf("fluent-bit service does not exist yet"))
```